lms/djangoapps/courseware/block_render.py

```diff
@@ -482,16 +561,6 @@ def inner_get_block(block):
             request_token=request_token,
         ))
 
-    # HACK: The following test fails when we do not access this attribute.
-    #  lms/djangoapps/courseware/tests/test_views.py::TestRenderXBlock::test_success_enrolled_staff
-    #  This happens because accessing `field_data` caches the XBlock's parents through the inheritance mixin.
-    #  If we do this operation before assigning `inner_get_block` to the `get_block_for_descriptor` attribute, these
-    #  parents will be initialized without binding the user data (that's how the `get_block` works in `x_module.py`).
-    #  If we retrieve the parents after this operation (e.g., in the `enclosing_sequence_for_gating_checks` function),
-    #  they will have their runtimes initialized, which can lead to an altered behavior of the XBlock-specific
-    #  parts of other runtimes. We will keep this workaround until we remove all XBlock-specific code from here.
-    block.static_asset_path  # pylint: disable=pointless-statement
```

Comment:
Fixed by dc3fd308a18ff02d69f2f7bdb9f501a174db759f. [This conversation](https://github.com/openedx/edx-platform/pull/8240#discussion_r31749841) explains why `disable_staff_debug_info` was added here. The `render_xblock` view is used in two places:
1. In the Learning MFE (as an iframe).
2. In the LTI context where Open edX is the LTI Provider.

It looks like the "staff debug info" button has been working correctly in the MFE by a pure accident - this parameter was overridden when the view initialized parent XBlocks.